luigi/worker.py

```diff
@@ -172,14 +172,25 @@ def run(self):
     def _recursive_terminate(self):
         import psutil
 
-        parent = psutil.Process(self.pid)
-        for child in parent.children(recursive=True):
-            child.kill()
+        try:
+            parent = psutil.Process(self.pid)
+            children = parent.children(recursive=True)
+
+            # terminate parent. Give it a chance to clean up
+            super(TaskProcess, self).terminate()
+            parent.wait()
```

Comment:
Wait, is it really a good idea to block here? What if it's a process that never terminates even though we've "asked for it"? Won't the whole worker-loop block on this then? @ChrisBeaumont